src/preconnect.js

```diff
@@ -382,3 +383,20 @@ export function preconnectForElement(element) {
   const preconnectService = getService(serviceHolder, 'preconnect');
   return new Preconnect(preconnectService, element);
 }
+
+/**
+ * Preconnects to the source URL and canonical domains to make sure
+ * outbound navigations are quick. Waits for onload to avoid blocking
+ * more high priority loads.
+ * @param {!Document} document
+ * @return {Promise} When work is done.
+ */
+export function preconnectToOrigin(document) {
+  return whenDocumentComplete(document).then(() => {
+    const element = document.documentElement;
+    const preconnect = preconnectForElement(element);
+    const info = Services.documentInfoForDoc(element);
+    preconnect.url(info.sourceUrl);
+    preconnect.url(info.canonicalUrl);
```

Comment:
Shouldn't this be waiting for `whenViewerVisible`? I think this will leak the user's search before click without it.

Reply:
Preconnect does this internally.